gmailRest/gmailRest-connector/gmailRest-connector-1.0.0/org.wso2.carbon.connector/src/main/java/org/wso2/carbon/connector/GmailUtils.java

```diff
@@ -831,4 +833,97 @@ private static void addAttachmentToMessageContext(String attachmentContentID,
         axis2mc.addAttachment(attachmentContentID, handler);
         log.info("Added an attachemnt named \"" + attachmentContentID + "\" to message context");
     }
+
+
+    public static boolean validateToken (String accessToken) {
+        String validationUrl = "https://www.googleapis.com/oauth2/v1/tokeninfo?access_token=" + accessToken;
+        try {
+            URL urlObj = new URL(validationUrl);
+            HttpsURLConnection httpsURLConnection = (HttpsURLConnection) urlObj.openConnection();
+
+            httpsURLConnection.setRequestMethod("GET");
+            if (httpsURLConnection.getResponseCode() == 400){
+                return false;
+            }
+            else {
+                return true;
+            }
+
+        } catch (MalformedURLException e) {
+            System.out.println("Failed to validate access Token");
```

Comment:
out out message and exception not relevant. message should give idea about exception.